src/components/Post.js

```diff
@@ -50,19 +61,47 @@ class Post extends React.Component {
     commentId: string
   }
 
+  static contextTypes = {currentUser: object}
+
   static childContextTypes = {post: object}
 
   getChildContext () {
     return {post: this.props.post}
   }
 
   render () {
-    const { post, comments, expanded, onExpand, community } = this.props
+    let  { post } = this.props
+    const { comments, expanded, onExpand, community, dispatch, contributorChoices } = this.props
+    const { contributors } = this.state
+    const { currentUser } = this.context
     const { communities, tag, media, linkPreview } = post
     const image = find(m => m.type === 'image', media)
     const classes = cx('post', tag, {image, expanded})
     const title = linkifyHashtags(decode(sanitize(post.name || '')), get('slug', community))
 
+    const canEdit = canEditPost(currentUser, post)
+
+    const isRequest = post.tag === 'request'
+    const isCompleteRequest = isRequest && post.fulfilled_at
+    const isIncompleteRequest = isRequest && !post.fulfilled_at && canEdit
+    const toggleRequestCompleting = () => {
+      return this.setState({requestCompleting: !this.state.requestCompleting})
+    }
+    const addContributor = (person) => {
+      this.setState({contributors: [...this.state.contributors, person]})
+    }
+    const removeContributor = (person) => {
+      this.setState({contributors: reject(this.state.contributors, {id: person.id})})
+    }
+    const contributorHandleInput = (term) => {
+      dispatch(typeahead(term, 'invite', {communityId: community.id, type: 'people'}))
+    }
+    const completeRequest = () => {
+      if(contributors.length > 0) {
```

Comment:
Be good to either allow completion with no contributors, or if not, add a notify saying "please choose at least one contributor"